app/models/payments/ChargeType.scala

```diff
@@ -0,0 +1,146 @@
+  override val value: String = "VAT AA Default Interest"
+}
+case object AAFurtherInterestCharge extends ChargeType {
+  override val value: String = "VAT AA Further Interest"
+}
+case object AACharge extends ChargeType {
+  override val value: String = "VAT Additional Assessment"
+}
+case object BnpRegPre2010Charge extends ChargeType {
+  override val value: String = "VAT BNP of Reg Pre 2010"
+}
+case object BnpRegPost2010Charge extends ChargeType {
+  override val value: String = "VAT BNP of Reg Post 2010"
+}
+case object FtnMatPre2010Charge extends ChargeType {
+  override val value: String = "VAT FTN Mat Change Pre 2010"
+}
+case object FtnMatPost2010Charge extends ChargeType {
+  override val value: String = "VAT FTN Mat Change Post 2010"
+}
+case object FtnEachPartnerCharge extends ChargeType {
+  override val value: String = "VAT FTN Each Partner"
+}
+case object MiscPenaltyCharge extends ChargeType {
+  override val value: String = "VAT Miscellaneous Penalty"
+}
+case object MpPre2009Charge extends ChargeType {
+  override val value: String = "VAT MP pre 2009"
+}
+case object MpRepeatedPre2009Charge extends ChargeType {
+  override val value: String = "VAT MP (R) pre 2009"
+}
+case object CivilEvasionPenaltyCharge extends ChargeType {
+  override val value: String = "VAT Civil Evasion Penalty"
+}
+
+object ChargeType {
+
+  def apply: String => ChargeType = {
+    case ReturnCharge.value => ReturnCharge
+    case ReturnDebitCharge.value => ReturnDebitCharge
+    case ReturnCreditCharge.value => ReturnCreditCharge
+    case OACharge.value => OACharge
+    case OACreditCharge.value => OACreditCharge
+    case OADebitCharge.value => OADebitCharge
+    case DefaultSurcharge.value => DefaultSurcharge
+    case CentralAssessmentCharge.value => CentralAssessmentCharge
+    case ErrorCorrectionCharge.value => ErrorCorrectionCharge
+    case ErrorCorrectionCreditCharge.value => ErrorCorrectionCreditCharge
+    case ErrorCorrectionDebitCharge.value => ErrorCorrectionDebitCharge
+    case RepaymentSupplement.value => RepaymentSupplement
+    case RepaySupplement.value => RepaySupplement
+    case OADefaultInterestCharge.value => OADefaultInterestCharge
+    case AACharge.value => AACharge
+    case AAInterestCharge.value => AAInterestCharge
+    case AAFurtherInterestCharge.value => AAFurtherInterestCharge
+    case BnpRegPre2010Charge.value => BnpRegPre2010Charge
+    case BnpRegPost2010Charge.value => BnpRegPost2010Charge
+    case FtnEachPartnerCharge.value => FtnEachPartnerCharge
+    case FtnMatPost2010Charge.value => FtnMatPost2010Charge
+    case FtnMatPre2010Charge.value => FtnMatPre2010Charge
+    case MiscPenaltyCharge.value => MiscPenaltyCharge
+    case MpPre2009Charge.value => MpPre2009Charge
+    case MpRepeatedPre2009Charge.value => MpRepeatedPre2009Charge
+    case CivilEvasionPenaltyCharge.value => CivilEvasionPenaltyCharge
+    case _ => throw new RuntimeException("Invalid Charge Type")
```

Comment:
You could potentially use `IllegalArgumentException` here, just to be more specific